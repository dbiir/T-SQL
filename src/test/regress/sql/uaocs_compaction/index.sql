-- @Description Tests basic index usage behavior after vacuuming
CREATE TABLE uaocs_index (a INT, b INT, c CHAR(128)) WITH (appendonly=true, orientation=column);
CREATE INDEX uaocs_index_index ON uaocs_index(b);

INSERT INTO uaocs_index SELECT i as a, i as b, 'hello world' as c FROM generate_series(1,10) AS i;
INSERT INTO uaocs_index SELECT i as a, i as b, 'hello world' as c FROM generate_series(1,10) AS i;

VACUUM uaocs_index;
SELECT * FROM uaocs_index WHERE b = 5;
