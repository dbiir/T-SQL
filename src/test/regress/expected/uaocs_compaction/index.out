-- @Description Tests basic index usage behavior after vacuuming
CREATE TABLE uaocs_index (a INT, b INT, c CHAR(128)) WITH (appendonly=true, orientation=column);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'a' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
CREATE INDEX uaocs_index_index ON uaocs_index(b);
INSERT INTO uaocs_index SELECT i as a, i as b, 'hello world' as c FROM generate_series(1,10) AS i;
INSERT INTO uaocs_index SELECT i as a, i as b, 'hello world' as c FROM generate_series(1,10) AS i;
VACUUM uaocs_index;
SELECT * FROM uaocs_index WHERE b = 5;
 a | b |                                                                c                                                                 
---+---+----------------------------------------------------------------------------------------------------------------------------------
 5 | 5 | hello world                                                                                                                     
 5 | 5 | hello world                                                                                                                     
(2 rows)

